sound/soc/intel/boards/sof_sdw.c

```diff
@@ -253,12 +253,21 @@ static struct sof_sdw_codec_info codec_info_list[] = {
 	},
 };
 
-static inline int find_codec_info_part(unsigned int part_id)
+static inline int find_codec_info_part(u64 adr)
 {
+	unsigned int part_id, sdw_version;
 	int i;
 
+	part_id = SDW_PART_ID(adr);
+	sdw_version = SDW_VERSION(adr);
 	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++)
-		if (part_id == codec_info_list[i].id)
+		/*
+		 * A codec info is for all sdw version with the part id if
+		 * v_id is not specified in the codec info.
+		 */
+		if (part_id == codec_info_list[i].id &&
+		    (!codec_info_list[i].v_id ||
+		     sdw_version == codec_info_list[i].v_id))
 			break;
```

Comment:
in fact you could just `return i;` here and `return -EINVAL;` after the loop unconditionally, maybe this is what @ranj063 meant?

Reply:
> in fact you could just `return i;` here and `return -EINVAL;` after the loop unconditionally, maybe this is what @ranj063 meant?

Fixed